### PR TITLE
Feature: (ISA-161) Habitat observations queries

### PR DESCRIPTION
**Pull request by JoshGx2000**
[ISA-161](https://jira.its.utas.edu.au/browse/ISA-161)

~~Habitat observations in backend and used by frontend to update the app state. SQUIDLE observations query in the backend will need changes shortly when Emma comes through on some column updates.~~
Edit: SQUIDLE stuff has been updated and the habitat observations queries work as intended 🙂

### CODE REVIEW

File: backend/habitat/viewsets.py [hunk 1]
```diff
@@ -870,3 +1083,82 @@ def bathymetry_statistics(request):
             return Response([])
         else:
             return Response(bathymetry_stats)
+
+@action(detail=False)
+@api_view()
+@renderer_classes((JSONRenderer, ShapefileRenderer))
```
Comment (markhepburn) [inline backend/habitat/viewsets.py:1089]: It's not a bad idea, but is there a concrete reason for ShapefileRenderer to be included here?  (more specifically, have you tested it?)

File: backend/habitat/viewsets.py [hunk 2]
```diff
@@ -870,3 +1083,82 @@ def bathymetry_statistics(request):
...
+            raise Exception('Unhandled boundary type!')
+
+        try:
+            columns = [col[0] for col in cursor.description]
+            namedrow = namedtuple('Result', columns)
+            results = [namedrow(*row) for row in cursor.fetchall()]
+
+            global_archive = [row._asdict() for row in results]
+        except:
+            pass
```
Comment (markhepburn) [inline backend/habitat/viewsets.py:1124]: What's the thinking behind this?  Do we expect benign exceptions that we ignore, or is this just convenient?

(I'm not against it per-se, just wondering if any thought has gone into it, or it was just convenient at the time)

**Review by markhepburn — COMMENTED** [review]
Looks good in general, just a couple of comments I left about general practice.

File: backend/habitat/viewsets.py (hunk 1 above)
Comment (JoshGx2000) [reply backend/habitat/viewsets.py:1089]: No reason, just a copy+paste error

File: backend/habitat/viewsets.py (hunk 2 above)
Comment (JoshGx2000) [reply backend/habitat/viewsets.py:1124]: Yeah, it's probably a good idea to not allow any benign exceptions. I wasn't really thinking. We'll just have it fail on any exception 👍

**Pull request merged**